samples/dnn/text_detection.cpp

```diff
@@ -159,3 +220,71 @@ void decode(const Mat& scores, const Mat& geometry, float scoreThresh,
         }
     }
 }
+
+void four_points_transform(const Mat& frame, const RotatedRect& box, const Size2f& box_size, const Point2f& ratio,
+                           Mat& result) {
+
+    Point2f vertices[4];
+    box.points(vertices);
+    for (int j = 0; j < 4; ++j)
+    {
+        vertices[j].x *= ratio.x;
+        vertices[j].y *= ratio.y;
+    }
+
+
+    Point2f target_vertices[4] = {Point2f(0, box_size.height - 1),
+                                  Point2f(0, 0), Point2f(box_size.width - 1, 0),
+                                  Point2f(box_size.width - 1, box_size.height - 1),
+                                  };
+    // Compute width
+    int width_a = std::sqrt(std::pow(target_vertices[1].x - target_vertices[2].x, 2)
+                + std::pow(target_vertices[2].y - target_vertices[3].y, 2));
+
+    int width_b = std::sqrt(std::pow(target_vertices[0].x - target_vertices[3].x, 2)
+                + std::pow(target_vertices[0].y - target_vertices[3].y, 2));
+
+    int max_width = std::max(width_a, width_b);
+
+    // Compute height
+    int height_a = std::sqrt(std::pow(target_vertices[0].x - target_vertices[1].x, 2)
+                + std::pow(target_vertices[0].y - target_vertices[1].y, 2));
+
+    int height_b = std::sqrt(std::pow(target_vertices[2].x - target_vertices[3].x, 2)
+                + std::pow(target_vertices[2].y - target_vertices[3].y, 2));
+
+    int max_height = std::max(height_a, height_b);
+
+    Mat rotation_matrix = getPerspectiveTransform(vertices, target_vertices);
+    warpPerspective(frame, result, rotation_matrix, Size(max_width, max_height));
+}
+
+void decode_text(const Mat& scores, std::string& text) {
+    static const std::string alphabet = "0123456789abcdefghijklmnopqrstuvwxyz";
+    std::vector<int> dims = {scores.size[0], 37};
+    Mat scores_mat = scores.reshape(1, dims);
+
+    std::vector<char> elements;
+    elements.reserve(scores.size[0]);
+
+
+    for (int row_index = 0; row_index < scores_mat.rows; ++row_index) {
+        Point p;
+        minMaxLoc(scores_mat.row(row_index), 0, 0, 0, &p);
+        if (p.x > 0) {
+            elements.push_back(alphabet[p.x - 1]);
+        } else {
+            elements.push_back('-');
+        }
+    }
+
+    if (elements.size() > 0 && elements[0] != '-') {
+        text += elements[0];
+    }
+    for (size_t element_index = 1; element_index < elements.size(); ++element_index) {
+        if (element_index > 0 && elements[element_index] != '-' &&
+            elements[element_index - 1] != elements[element_index]) {
+            text += elements[element_index];
+        }
+    }
```

Comment:
So maybe we just don't need to push `'-'` characters  to the `elements`?

Reply:
This function is correct only because let's say the original text is 'aaaa'
and the model predicted "a--aa--aaa----a"
if we remove the '-'s (aaaaaaa) and then we remove the adjacent same letters then we will be left with only 'a'
whereas if we use the current method we will get the correct output "aaaa"

Reply:
@themechanicalcoder, thanks for example. @rngtna, please check if current implementation will return "aaaa" for "a--aa--aaa----a".

Reply:
I checked it works correctly